probe/sniff/sniffer_test.go

```diff
@@ -77,7 +72,16 @@ func TestMerge(t *testing.T) {
 			},
 		},
 		NodeMetadatas: report.NodeMetadatas{
-			srcEndpointNodeID: report.MakeNodeMetadata(),
+			srcEndpointNodeID: {
+				Metadata:  map[string]string{},
+				Counters:  map[string]int{},
+				Adjacency: report.MakeIDList(dstEndpointNodeID),
+			},
+			dstEndpointNodeID: report.NodeMetadata{
+				Metadata:  map[string]string{},
+				Counters:  map[string]int{},
+				Adjacency: report.MakeIDList(),
+			},
```

Comment:
I've never been a fan of that builder style of constructor, but it may be the least-bad option in this case...

```go
md := report.MakeNodeMetadata().SetMetadata(map[string]string{
    "foo": "bar",
    "baz": "quux",
}).SetAdjacency(report.MakeIDList(
    dstEndpointNodeID,
))
```

![okay](http://i.imgur.com/HAM7M8U.jpg)